pylot/drivers/carla_obstacles_driver_operator.py

```diff
@@ -0,0 +1,48 @@
+import carla
+
+import erdos
+
+import pylot.utils
+from pylot.drivers.carla_base_gnss_driver_operator import (
+    CarlaBaseGNSSDriverOperator)
+from pylot.perception.detection.obstacle import Obstacle
+from pylot.perception.messages import ObstaclesMessage
+
+
+class CarlaObstaclesDriverOperator(CarlaBaseGNSSDriverOperator):
+    """Publishes the bounding boxes of all vehicles and people retrieved from
+    the simulator at the provided frequency.
```

Comment:
Why were these operators subclassed from a `BaseGNSSDriverOperator`? They don't seem to be using the `gnss_msg` exposed in the `process_gnss` method and retrieve different data from the simulator.

Reply:
My thought process is the following:
1. The original CARLA operator performs complicated logic to send data from the simulator (e.g. pose, ground obstacles, etc) at multiple frequencies (localization and control). This logic is coupled with how the operator ticks the simulator.
2. Sensors can define tick frequencies independently. Therefore, we can set up a lightweight sensor such as GNSS which calls the logic for sending arbitrary information at arbitrary frequencies.
3. The logic for setting up the GNSS sensor is identical across several operators, so I moved the shared logic to `BaseGNSSDriverOperator`. I could also move it to a helper method or rename `process_gnss`, if you think that's cleaner.

I checked if any other actors can invoke callbacks at configurable frequencies, but it appears that this is limited to sensors.

Reply:
So the GNSS sensor is being used as a timestamping mechanism? Why not just have a base operator that sends its downstream operators a watermark message to retrieve and release data?

Reply:
That works too, but it would still share logic with the GNSS sensor operator, and we'd still need to instantiate one for each downstream operator frequency. So the question is whether the data release mechanism should be a standalone operator or a coupled with the driver operators.

Personally, I think it makes sense to put the logic in the driver operator so that the API reflects the other driver operators and simplifies pipeline architecture.